components/channel_select/channel_select.jsx

```diff
@@ -72,14 +37,18 @@ export default class ChannelSelect extends React.Component {
             </option>,
         ];
 
-        this.state.channels.forEach((channel) => {
+        this.props.channels.forEach((channel) => {
+            let channelName = channel.display_name;
+            if (!channelName || channelName.length === 0) {
```

Comment:
Minor nit, but this a bit redundant since `!channelName` is true if `channelName` is a string and has a length of 0. The only way that this would make any difference would be if channelName could be an array. This could also be shortened to `const channelName = channel.display_name || channel.name`